mk/aosp_optee.mk

```diff
@@ -19,42 +19,47 @@
 ##########################################################
 ## define common variables, like TA_DEV_KIT_DIR         ##
 ##########################################################
-OPTEE_OUT_DIR ?= $(realpath $(PRODUCT_OUT))/optee
+TOP_ROOT_ABS := $(realpath $(TOP))
+
+# OPTEE_OUT_DIR could be exported explicitly
+# if PRODUCT_OUT is not the default out directory in aosp workspace
+OPTEE_OUT_DIR ?= $(PRODUCT_OUT)/optee
+ABS_OPTEE_OUT_DIR ?= $(realpath $(PRODUCT_OUT))/optee
 OPTEE_TA_OUT_DIR ?= $(OPTEE_OUT_DIR)/ta
+ABS_OPTEE_TA_OUT_DIR ?= $(ABS_OPTEE_OUT_DIR)/ta
 # Set so that OP-TEE clients can find the installed dev-kit, which
 # depends on platform and its OP-TEE word-size.
 OPTEE_OS_OUT_DIR ?= $(OPTEE_OUT_DIR)/arm-plat-$(OPTEE_PLATFORM)
-TA_DEV_KIT_DIR := $(OPTEE_OS_OUT_DIR)/export-${OPTEE_TA_TARGETS}
+ABS_OPTEE_OS_OUT_DIR := $(ABS_OPTEE_OUT_DIR)/arm-plat-$(OPTEE_PLATFORM)
+TA_DEV_KIT_DIR := $(ABS_OPTEE_OS_OUT_DIR)/export-${OPTEE_TA_TARGETS}
 
-TOP_ROOT_ABS := $(realpath $(TOP))
 CROSS_COMPILE64 := $(TOP_ROOT_ABS)/$(TARGET_TOOLS_PREFIX)
 CROSS_COMPILE_LINE := CROSS_COMPILE64="$(CROSS_COMPILE64)"
 ifneq ($(strip $($(combo_2nd_arch_prefix)TARGET_TOOLS_PREFIX)),)
 CROSS_COMPILE32 := $(TOP_ROOT_ABS)/$($(combo_2nd_arch_prefix)TARGET_TOOLS_PREFIX)
 CROSS_COMPILE_LINE += CROSS_COMPILE32="$(CROSS_COMPILE32)"
 endif
 
-OPTEE_BIN := $(TOP_ROOT_ABS)/$(OPTEE_OS_OUT_DIR)/core/tee.bin
+OPTEE_BIN := $(OPTEE_OS_OUT_DIR)/core/tee.bin
 
-$(OPTEE_BIN) : BUILD_OPTEE_OS
+$(OPTEE_BIN) : $(sort $(shell find -L $(OPTEE_OS_DIR)))
```

Comment:
Why find? Why sort? Why not use $(OPTEE_OS_DIR) directly?

Reply:
I do like this according to the suggestion from here:
https://android.googlesource.com/platform/build/+/master/Changes.md#and-calling-other-build-systems

The $(OPTEE_BIN) here is the same as the zImage building there, and the explanations is "to preserve the speed of Android incrementals"

Reply:
Oh, right, I see, `$(OPTEE_OS_DIR)` is the source directory. That's fine.

Thanks for the explanation. For this patch too:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`